File: 2018/day02/part2.py
from collections import defaultdict

def solve(lines, offset=0):
    sublines = set()
    for line in lines:
        subline = line[:offset] + line[offset+1:]
        if subline in sublines:
            return subline
        sublines.add(subline)

with open('input.txt', 'r') as f:
    lines = [l for l in f]

    for i in range(0, len(lines[0])):
        result = solve(lines, i)
        if result:
            print("Answer is: {}".format(result))
            break
    else:
        print("No answer found?")

File: 2018/day02/part2.clj
(use '[clojure.string :only (join)])

(defn omit-char [line index]
  (join [(subs line 0 index)
         (subs line (+ index 1))]))

(defn get-input []
  (str/split (slurp "input.txt") #"\n"))

(defn for-offset [lines index]
  (loop [seen #{}
         head (first lines)
         tail (rest lines)]
    (let [calcd (omit-char head index)]
      if (contains? seen head) head
        (recur (conj seen head) (first tail) (rest tail)))))

(defn solve []
  (let [lines (get-input)
        line_length (count (get lines 0))
        offsets (range line_length)]
    (take-while (partial for-offset lines) offsets)))

(println solve)

